src/recipeService.ts

```diff
@@ -0,0 +1,94 @@
+import { Recipe } from "./recipe";
+import mongoose, { Schema } from "mongoose";
+
+export type RecipeCreationParams = Pick<
+  Recipe,
+  | "id"
+  | "name"
+  | "thumbnail"
+  | "description"
+  | "link"
+  | "metadata"
+  | "ingredients"
+  | "steps"
+>;
+
+const url = process.env.DB_URI;
+
+mongoose.connect(url);
+const db = mongoose.connection;
+db.on("error", (error) => {
+  console.log(error);
+});
+db.once("connected", () => {
+  console.log("Database Connected");
+});
+
+var recipeSchema = new Schema({
+  id: Number,
+  name: String,
+  thumbnail: String,
+  description: String,
+  link: [],
+  metadata: {},
+  ingredients: [],
+  steps: [],
+});
+
+const recipeModel = mongoose.model("recipes", recipeSchema);
+
+export async function getRecipe(id: number) {
+  try {
+    return await recipeModel.find({ id: id });
```

Comment:
This should be findOne, because find retruns an array and thats not what you want here